github/pulls.go

```diff
@@ -301,11 +301,11 @@ type pullRequestMergeRequest struct {
 func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string, options *PullRequestOptions) (*PullRequestMergeResult, *Response, error) {
 	u := fmt.Sprintf("repos/%v/%v/pulls/%d/merge", owner, repo, number)
 
-	pullRequestBody := &pullRequestMergeRequest{CommitMessage: &commitMessage}
+	pullRequestBody := pullRequestMergeRequest{CommitMessage: commitMessage}
```

Comment:
I don't agree with making `pullRequestBody` a value rather than a pointer.

Although this code is typically not performance critical, it is not necessary to make copies of structs (and therefore make copies of its members) on function calls and I think we want to demonstrate the more efficient version.

Additionally, this change would be inconsistent with the rest of this file using pointers, and more generally with the `go-github` package.